src/routes/rooms.ts

```diff
@@ -32,53 +33,31 @@ router.get(
     });
 
     if (!room) {
-      throw new NotFoundError(ERROR_ROOM_GET_BY_ID, 'Room does not exists');
+      throw new NotFoundError(
+        ERROR_ROOM_GET_BY_ID,
+        ERROR_ROOM_GET_BY_ID_MESSAGE,
+      );
     } else {
       res.send({ data: room });
     }
   }),
 );
 
-// TODO: discuss if it's fine that types are verbs.
-type CreateRoom = {
-  name?: string;
-  description?: string;
-  options: Array<{
-    label?: string;
-    value?: string;
-  }>;
-};
-
 router.post(
   '/',
   asyncHandler(async (req, res) => {
-    const body: CreateRoom = req.body;
+    const newRoom: RoomInterface = req.body;
+    const { status, error, message } = validateRoom(newRoom);
 
-    if (!body) {
-      throw new UnprocessableEntityError(
-        ERROR_ROOM_CREATE_EMPTY_DATA,
-        'Empty body sent',
-      );
+    if (status === ValidationRoomStatus.INVALID) {
+      throw new UnprocessableEntityError(error, message);
     }
 
-    const hasOptions = body.options && Array.from(body.options).length > 0;
-    if (!hasOptions) {
-      throw new UnprocessableEntityError(
-        ERROR_ROOM_CREATE_EMPTY_OPTIONS,
-        'Options cannot be empty',
-      );
-    }
-
-    const room = new Room();
-    room.name = body.name;
-    room.description = body.description;
-    room.options = body.options.map(params => {
-      const option = new Option();
-      option.label = params.label;
-      option.value = params.value;
+    const room: Room = new Room();
+    room.name = newRoom.name;
+    room.description = newRoom.description || 'empty description';
+    room.options = newRoom.options.map(Option.New);
 
-      return option;
-    });
     const [validationError] = await validate(room);
```

Comment:
this is the validation I was talking about

Reply:
I'm confused. Where the validation is declared here?

Reply:
you'll need to check the file, but this validation is not included in the abstraction you made